src/joyride/js_completer.cljs

```diff
@@ -0,0 +1,41 @@
+(ns joyride.js-completer
+  (:require [promesa.core :as p]
+            ["repl" :as node-repl]
+            ["vscode" :as vscode]))
+
+(defn complete+
+  "Returns a promise resolving to the completion results of `s` given a JavaScript `context`."
+  [s ^js context]
+  (p/create (fn [resolve reject]
+              (let [repl (.start node-repl)]
+                (.completer repl
+                            s
+                            (fn [error result]
+                              (if-not error
+                                (resolve (->> result
+                                              first
+                                              js->clj
+                                              (remove empty?)))
+                                (reject error))
+                              (.close repl))
+                            (.assign js/Object (.-context repl) context))))))
+
+(defn js-keys+
+  "Returns a promise resolving to all keys of `obj` as a Javascript array."
+  [^js obj]
+  (p/let [o-name (name (gensym))
+          prefix (str o-name ".")
+          completions+ (complete+ prefix (clj->js {o-name obj}))]
+    (->> completions+
+         (map (fn [completion]
```

Comment:
completions+ returns a promise. How does `map` work on a promise object?

Reply:
oh right, because of `p/let`

Reply:
It's unwrapped in the `p/let`, isn't it?